packages/core/src/mpi/filter-patients/filter-patients.ts

```diff
@@ -0,0 +1,129 @@
+import { normalizePatientInboundMpi } from "../normalize-patient";
+import { PatientData } from "../../domain/patient";
+import { calculateNameScore } from "./match-name";
+import { calculateDobScore } from "./match-dob";
+import { calculateGenderScore } from "./match-gender";
+import { calculateAddressScore } from "./match-address";
+import { calculateContactScores } from "./match-contact";
+import { calculateSsnScore, hasSsnData } from "./match.ssn";
+import { checkBusinessRules } from "./match-business-rules";
+import { crossValidateInvalidLinks } from "./cross-validate-links";
+
+type LinkStatus = {
+  patient: PatientData;
+  isMatch: boolean;
+  totalScore: number;
+  scores: { [key: string]: number };
+  failedRule: string | undefined;
+};
+
+export function filterPatientLinks(
+  metriportPatient: PatientData,
+  patientLinks: PatientData[],
+  threshold: number
+): LinkStatus[] {
+  let validLinks: LinkStatus[] = [];
+  let invalidLinks: LinkStatus[] = [];
+
+  for (const patientLink of patientLinks) {
+    const { isMatch, totalScore, scores, failedRule } = linkFilteringAlgorithm(
+      metriportPatient,
+      patientLink,
+      threshold
+    );
+
+    if (isMatch) {
+      validLinks.push({
+        patient: patientLink,
+        isMatch,
+        totalScore,
+        scores,
+        failedRule,
+      });
+    } else {
+      invalidLinks.push({
+        patient: patientLink,
+        isMatch,
+        totalScore,
+        scores,
+        failedRule,
+      });
+    }
+  }
+
+  const crossValidatedLinks = crossValidateInvalidLinks(
+    validLinks.map(link => link.patient),
+    invalidLinks.map(link => link.patient)
+  );
+
+  const crossValidatedLinkStatuses = invalidLinks
+    .filter(link => crossValidatedLinks.includes(link.patient))
+    .map(link => ({
+      ...link,
+      isMatch: true,
+      failedRule: undefined,
+    }));
+
+  validLinks = [...validLinks, ...crossValidatedLinkStatuses];
+  invalidLinks = invalidLinks.filter(link => !crossValidatedLinks.includes(link.patient));
+
+  return [...validLinks, ...invalidLinks];
+}
+
+export function linkFilteringAlgorithm(
```

Comment:
can we rename this to something a little more obvious? "evaluatePatientMatch" or something